ctfcli/core/challenge.py

```diff
@@ -143,9 +143,42 @@ def __init__(self, challenge_yml: Union[str, PathLike], overrides=None):
         # Set Image to None if the challenge does not provide one
         self.image = None
 
-        # Get name and a build path for the image if the challenge provides one
-        if self.get("image"):
-            self.image = Image(slugify(self["name"]), self.challenge_directory / self["image"])
+        # Create an image object, if the challenge provides one
+        challenge_image = self.get("image")
+        if challenge_image:
+            # if the image is a library image, prepend docker.io for a full URL
+            if challenge_image.startswith("library/"):
+                challenge_image = f"docker.io/{challenge_image}"
+
+            # check if the image name defines a registry
+            for registry in [
+                "docker.io",
+                "gcr.io",
+                "ecr.aws",
+                "ghcr.io",
+                "azurecr.io",
+                "registry.digitalocean.com",
+                "registry.gitlab.com",
+                "registry.ctfd.io",
+            ]:
```

Comment:
There should probably be a way to force registry detection. Some people may choose to use their own self hosted registry which wouldn't be detected with this information. 

Perhaps we force registry behavior if the image starts with `registry://` ?

Reply:
Sure, that makes sense to me!